src/hooks/useSelector.js

```diff
@@ -0,0 +1,138 @@
+import React, {
+  useContext,
+  useRef,
+  useMemo,
+  useCallback,
+  useLayoutEffect,
+  useEffect
+} from 'react'
+
+import { ReactReduxContext } from '../components/Context'
+
+// React currently throws a warning when using useLayoutEffect on the server.
+// To get around it, we can conditionally useEffect on the server (no-op) and
+// useLayoutEffect in the browser. We need useLayoutEffect to ensure we can
+// run effects before the node is updated in our queue
+const useIsomorphicLayoutEffect =
```

Comment:
for purely semantic reasons, I would rename this to `useLayoutEffectPlatformed`

Reply:
this is idiomatic in current react-redux so I'll leave as is since it is not a change I introduced